trading_cost.py

```diff
@@ -0,0 +1,57 @@
+"""
+A file to build cost objects.
+"""
+
+class CurrencyCost:
+    """
+    Cost function of currency trading.
+    """
+    def __init__(self, base_ins, instruments):
+        self.base_ins = base_ins
+        self.cost = 0
+        self.profit = 0
+        self.quantity = {
+            instrument : 0 for instrument in instruments}
+        self.value = {
+            instrument : 0 for instrument in instruments}
+        self.value[base_ins] = 1
+        self.quantity[base_ins] = 1
+
+    def __call__(self, portfolio, price):
+        self.reset()
+        self.calculate_profit(portfolio[-1], price)
+        return self.profit - self.cost, self.cost
+
+    def calculate_profit(self, today, price):
+        """
+        Calculate profit and produce cost.
+        """
+        total_value = 0
+        for instrument in today:
+            if price[instrument][-1] is not None:
+                total_value += today[instrument]
+                self.cal_single_ins_profit(
+                    today, instrument, price[instrument][-1])
+        base_value = 1 - total_value
+        self.profit += (self.value[self.base_ins] - base_value)
+        self.value[self.base_ins] = base_value
+        self.quantity[self.base_ins] = base_value
+
+    def cal_single_ins_profit(self, today, instrument, price):
```

Comment:
While this method can be merge with calculate_profit, it make it a little unclear.
And also, we may change the way to calculate profit and cost for each instrument, so separated it.